v3/reutlingen.py

```diff
@@ -0,0 +1,122 @@
+"""
+Copyright 2024 binary butterfly GmbH
+Use of this source code is governed by an MIT-style license that can be found in the LICENSE.txt.
+"""
+
+import csv
+import re
+from datetime import datetime, timezone
+from decimal import Decimal
+from enum import Enum
+from io import StringIO
+from typing import Any
+
+import pyproj
+from validataclass.dataclasses import validataclass
+from validataclass.exceptions import ValidationError
+from validataclass.validators import (
+    DataclassValidator,
+    EnumValidator,
+    IntegerValidator,
+    ListValidator,
+    NumericValidator,
+    StringValidator,
+)
+
+from common.base_converter import CsvConverter
+from common.exceptions import ImportParkingSiteException
+from common.models import ImportSourceResult
+from common.validators import StaticParkingSiteInput
+from common.validators.base_validators import ParkingSiteTypeInput
+from common.validators.fields.noneable import Noneable
+from util import SourceInfo
+
+
+class ReutlingenParkingSiteType(Enum):
+    PARKHAUS = 'parkhaus'
+    TIEFGARAGE = 'tiefgarage'
+    PARKFLAECHE = 'parkfläche'
+    P_R = 'p+r'
+
+    def to_parking_site_type_input(self) -> ParkingSiteTypeInput:
+        # TODO: find out more details about this enumeration for a proper mapping
+        return {
+            self.PARKHAUS: ParkingSiteTypeInput.CAR_PARK,
+            self.TIEFGARAGE: ParkingSiteTypeInput.UNDERGROUND,
+            self.PARKFLAECHE: ParkingSiteTypeInput.OFF_STREET_PARKING_GROUND,
+        }.get(self, ParkingSiteTypeInput.OTHER)
+
+
+class PointCoordinateTupleValidator(ListValidator):
+    PATTERN = r'POINT \(([-+]?\d+\.\d+) ([-+]?\d+\.\d+)\)'
+
+    def validate(self, input_data: Any, **kwargs) -> list:
+        input_match = re.match(self.PATTERN, input_data)
```

Comment:
This will throw an exception if the input is not a string -> check for typing before.

Reply:
As any casted string will run into issues anyway, it would be better if you use the method `_ensure_type` from parent class like this: `self._ensure_type(input_data, str)` instead of casting.

Reply:
You don't need `if not isinstance(input_data, str):` there :) Also, `_ensure_type()` always returns `None`, so you always will get `None` there: https://github.com/binary-butterfly/validataclass/blob/2ce4dbcb6cb60cea5919d32918641db800b1784f/src/validataclass/validators/validator.py#L78

The idea behind `_ensure_type` is to ensure a type, this means either it's this type or an ValidationError is thrown.

Reply:
Thanks for the clarification. I have now adjusted it.